### ISSUE TITLE
Rate of validation dataset ?

### ISSUE THREAD

**Issue by kamelbouyacoub** [thtrieu/darkflow#97]
Hello 

I readed many documents in the literatures and I saw that most of them said, that we need data for training and data for test validation, and my question is in the code where is the part that do the validation test. and how can we fix the rate between the test validation and the train data. For example if I have 20 images what is the rate of validation images and the train images 

Thanks,

**Comment by eugtanchik** [comment]
Hello,
I am also looking for that in the source code and I have the same issue. Seems like there is not validation during training process. As I know, usually validation fraction of training dataset is 0.3 or 0.2.

**Comment by thtrieu** [comment]
Good suggestions.

**Comment by Costyv95** [comment]
There are 2 methods of doing that. You can split the data set in train and validate sets inside the code or just send 2 separate data sets, one for train and one for validate when you call the flow module.

Anyway, in order to do that you should add some new parameters in  default.py file, then modify the functions _batch, parse and shuffle from data.py (both yolo and yolov2 folders) and modify the method train() in flow.py file(here you only have to run another batch (every iteration or once a number of iterations)  using the same tensorflow session, but without returning the train_op so you don't modify the weights). You can also add another  tf.summary.FileWriter for validation so you can visualize your validation loss graph using tensorboard.

I personally chose to send 2 different data sets . It was pretty straight forward. I hope I was clear enough.

**Comment by Ridhwanluthra** [comment]
did someone add this functionality?

**Comment by denissato** [comment]
How the hyperparameters are adjusted during the training without validation in darkflow? I trained a dataset with 2,000 images (2 classes) in darkflow (tiny-yolov2) and darknet (tiny-yolov3). I used the instructions provided in each repository. Now I need to explain this in my graduation final-work and I'm not found information about how this work in darkflow. Someone could indicate me material or help with explaining please?
